src/AeroSharp/DataAccess/KeyValueAccess/KeyValueStore.cs

```diff
@@ -166,14 +166,23 @@ private T GetValueToWrite<T>(Func<T> addValueFunc, Func<T, T> updateValueFunc, I
         {
             // Adding this function in as a helper method to AddOrUpdate
             // Intention is to make the code in the retry simple and readable
-
             T result;
             if (value.First().Value is object)
             {
-                // modify the record's value by first mapping the record to the object and then passing that
-                // to the updateValueFunction
-                // returns the object to write down to the store
-                result = updateValueFunc(MapRecordsToTuple<T>(value, bin).First().Value);
+                T cachedValue = MapRecordsToTuple<T>(value, bin).First().Value;
+
+                if (!EqualityComparer<T>.Default.Equals(cachedValue, default(T)))
```

Comment:
Will this cause issues for folks who have implemented a custom `EqualityComparer` for `T`?